lib/backend/k8s/conversion_test.go

```diff
@@ -245,12 +299,20 @@ var _ = Describe("Test NetworkPolicy conversion", func() {
 		Expect(pol.Value.(*model.Policy).Selector).To(Equal(
 			"calico/k8s_ns == 'default' && label == 'value' && label2 == 'value2'"))
 		protoTCP := numorstring.ProtocolFromString("tcp")
-		Expect(pol.Value.(*model.Policy).InboundRules).To(ConsistOf(model.Rule{
-			Action:      "allow",
-			Protocol:    &protoTCP, // Defaulted to TCP.
-			SrcSelector: "calico/k8s_ns == 'default' && k == 'v' && k2 == 'v2'",
-			DstPorts:    []numorstring.Port{numorstring.SinglePort(80)},
-		}))
+		Expect(pol.Value.(*model.Policy).InboundRules).To(ConsistOf(
+			model.Rule{
+				Action:      "allow",
+				Protocol:    &protoTCP, // Defaulted to TCP.
+				SrcSelector: "calico/k8s_ns == 'default' && k == 'v' && k2 == 'v2'",
+				DstPorts:    []numorstring.Port{numorstring.SinglePort(80)},
+			},
+			model.Rule{
+				Action:      "allow",
+				Protocol:    &protoTCP, // Defaulted to TCP.
+				SrcSelector: "calico/k8s_ns == 'default' && k == 'v' && k2 == 'v2'",
+				DstPorts:    []numorstring.Port{numorstring.NamedPort("foo")},
```

Comment:
Could we have instead just one unified model.Rule?
```
model.Rule{
	Action:      "allow",
	Protocol:    &protoTCP, // Defaulted to TCP.
	SrcSelector: "calico/k8s_ns == 'default' && k == 'v' && k2 == 'v2'",
	DstPorts:    []numorstring.Port{numorstring.SinglePort(80), numorstring.NamedPort("foo")},
}
```
Is there a good reason why the converter does not do that?

Reply:
Yes, we could make that optimisation but it's non-trivial due to the need to split out TCP/UDP ports into separate rules.